### DISCUSSION TITLE
EStop doesn`t work on T41U5XBB

### DISCUSSION THREAD

**Discussion by Switch-Witch** [terjeio/ioSender#412] [Q&A]
Hi, I wired my emergency button (closes when pressed
) to the "Halt" Input on the T41U5XBB but it seems to have no effect when pressing or releasing the button. What could be wrong and how to get the emergency button as an estop to work?

best
Kai

**Comment by phil-barrett** [comment]
What is $14 set to? Have you verified with a DMM that the switch is working? What does the Signals Group in ioSender show you when try the switch?

**Reply by Switch-Witch** [reply]
Hi Phil, I tested it directly with a DMM when noticing that it is not working. In the signal group is no reaction when pressed the button. The status of $14=66. Do you have any ideas?

**Comment by phil-barrett** [comment]
For an N.O. switch, that setting should work. Do the other inputs (Cycle Start, Feed Hold, ...) work?  Have you tried using a wire to short Halt to Ground?

**Reply by terjeio** [reply]
Check that the signal arrives at the Teensy board. One of my boards had a faulty optocoupler.
If compiling yourself then also check that you have selected the correct board, T41BB5X_Pro and T41U5XBB uses different pins for e-stop.

**Reply by Switch-Witch** [reply]
Hi Terjeio, this could possibly be the reason - perhaps I chose the wrong board in the web builder... I will check it today. Thx a lot!

**Reply by Switch-Witch** [reply]
It was exactly this reason - I recompiled the firmware with T41U5XBB and it works - thank you very much!
